File: functions/dynamoFormatter.py
import re

def format_dynamodb_json(input_filename, output_filename):
    try:
        with open(input_filename, 'r') as file:
            content = file.read()
        
        # Replace all instances of "}\n" (end of a JSON object followed by a newline)
        # with "},\n" to separate the objects correctly.
            
        # This assumes each JSON object is on a single line.
        formatted_content = re.sub(r'\}\n', '},\n', content)
        
        # Remove any trailing comma which may exist after the last JSON object
        formatted_content = re.sub(r',\n$', '\n', formatted_content)
        
        # Enclose the entire content within square brackets to form a valid JSON array
        if not (formatted_content.startswith('[') and formatted_content.endswith(']')):
            final_content = f'[{formatted_content.rstrip(",\n")}]'
        else:
            final_content = formatted_content
        
        # Save the formatted content to a new file
        with open(output_filename, 'w') as file:
            file.write(final_content)
        
        print("File has been successfully formatted and saved.")
        
    except FileNotFoundError:
        print(f"File {input_filename} not found.")
    except Exception as e:
        print(f"An error occurred: {e}")

File: functions/dynamo-formatter.mjs
const formatDynamoDBJson = (inputFilename, outputFilename, fs) => {
  try {
    console.log("Correcting your dynamoJSON file to proper json :) ");
    let content = fs.readFileSync(inputFilename, { encoding: "utf8" });
    // Replace all instances of "}\n" with "},\n" to separate the objects correctly
    let formattedContent = content.replace(/\}\n/g, "},\n");

    // Remove any trailing comma which may exist after the last JSON object
    formattedContent = formattedContent.replace(/,\n$/, "\n");

    // Enclose the entire content within square brackets to form a valid JSON array
    if (!formattedContent.startsWith("[") || !formattedContent.endsWith("]")) {
      formattedContent = `[${formattedContent.trim().replace(/,$/, "")}]`;
    }

    // Save the formatted content to a new file
    fs.writeFileSync(outputFilename, formattedContent, { encoding: "utf8" });

    console.log("File has been successfully corrected & formatted and saved.");
  } catch (error) {
    if (error.code === "ENOENT") {
      console.log(`File ${inputFilename} not found.`);
    } else {
      console.log(`An error occurred: ${error}`);
    }
  }
};

export default formatDynamoDBJson;
